### PR TITLE
fix(capture): route flag events through capture + papercuts

### PR DESCRIPTION
**Pull request by eli-r-ph**
## Summary
Follow-on that needed to land prior to un-gating capture v1, and misc. papercuts identified in smoke testing.

- **`$feature_flag_called` routes through v1** when `capture-v1` is enabled
    - This previously went around the std SDK capture path
    - v1 support stays fire-and-forget with a **single attempt and no retry loop** on both async and blocking hosts
- **Accept any 2xx as success** in `v1_capture::after_response` (was exactly 200), so a future 201/202/207 from capture is not misclassified as a connection error. Malformed bodies on 2xx still surface as `Error::Serialization`.
- **Canonical SDK identity**: `posthog-sdk-info` / `user-agent` now send `posthog-rs/<version>` (was `posthog-rust/<version>`)
    - The name segment now matches the `$lib` value the v0 path sends, so  v1 backend attributes correctly in SDK Health, usage reports, and Library columns
- **CI**:  ensure full matrix of v1-relevant Rust build features are exercised in CI unit test runs

## Test plan

- [x] `cargo test` (async, v0): 183 pass
- [x] `cargo test --features capture-v1`: 226 pass
- [x] `cargo test --no-default-features` (blocking, v0): 185 pass
- [x] `cargo test --no-default-features --features capture-v1`: 225 pass
- [x] `cargo test --workspace`, `cargo fmt --check`, `cargo clippy -- -D warnings`
- [x] New tests: 2xx-range unit + e2e (201), sdk-info format pin (unit + e2e), `build_flag_event_request` (batch shape, defaults, compression), v1 flag-event routing + single-attempt-failure (async + blocking)

### CODE REVIEW

File: src/client/v1_capture.rs [hunk 1]
```diff
@@ -584,4 +620,149 @@ mod tests {
...
+                &mut pending,
+                &mut final_results,
+            );
+            assert!(
+                matches!(step, Step::Done),
+                "HTTP {status} should be treated as success"
+            );
+            assert_eq!(final_results.len(), 1);
+        }
+    }
```
[the remark this answers is not in the document]
Comment (eli-r-ph) [reply src/client/v1_capture.rs:642]: Repo style is loop-over-cases for status-code sweeps (see `src/client/retry.rs`, `tests/test_v0_retry.rs`, `tests/test_v1_capture.rs`, `tests/test_blocking.rs`) and the crate has no rstest dependency. The assert message includes the failing status, so the regressed case is identified in the panic output. Keeping the loop.

File: src/client/v1_capture.rs [hunk 2]
```diff
@@ -183,7 +206,9 @@ pub(crate) fn after_response(
     pending: &mut Vec<V1Event>,
     final_results: &mut HashMap<Uuid, EventResult>,
 ) -> Step {
-    if status == 200 {
+    // The backend sends exactly 200 today; accept the whole 2xx class so a
+    // future 201/202 isn't misclassified as a connection error.
+    if (200..=299).contains(&status) {
         let batch_resp: CaptureResponse = match serde_json::from_str(body) {
```
Comment (ioannisj) [inline src/client/v1_capture.rs:212]: Empty body would err here right? but there are some 2xx that traditionally have empty body. Probably not our case since from the comments I see we care about specific 2xx codes

**Review by ioannisj — APPROVED** [review]
LG with my very limited rs skills

File: .sampo/changesets/capture-v1-2xx-sdk-info-flag-events.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+---
```
Comment (jose-sequeira) [inline .sampo/changesets/capture-v1-2xx-sdk-info-flag-events.md:1]: Is this intentional?

**Review by jose-sequeira — APPROVED** [review]

File: src/client/v1_capture.rs (hunk 2 above)
Comment (eli-r-ph) [reply src/client/v1_capture.rs:212]: This should be fine atm, and sendBeacon (etc.) 204 response support isn't needed in this SDK...but as I think about it more, I want to check on another potential backend/frontend interaction corner case before I land this.

Thanks for mentioning it! 🙇

Comment (eli-r-ph) [reply src/client/v1_capture.rs:212]: Updated to shore up the empty body case and add a test for it in the v1 capture_batch codepath. Thanks again!
